File: src/main/java/sk/pa3kc/pojo/Texture.java
package sk.pa3kc.pojo;

import static  org.lwjgl.opengl.GL11.*;

public class Texture {
    private final int textureId;

    public Texture(int textureId) {
        this.textureId = textureId;
    }

    public int getTextureId() {
        return this.textureId;
    }

    public void bind() {
        glBindTexture(GL_TEXTURE_2D, this.textureId);
    }
}

File: src/main/kotlin/sk/pa3kc/poko/Texture.kt
package sk.pa3kc.poko

import org.lwjgl.opengl.GL11
import sk.pa3kc.util.GLBindable

data class Texture(
    val id: Int
) : GLBindable, AutoCloseable {
    override fun bind() = GL11.glBindTexture(GL11.GL_TEXTURE_2D, this.id)

    override fun close() {
        GL11.glDeleteTextures(this.id)
    }
}
